### PR TITLE
TableField: improve validation result and make it extensible

### PR DESCRIPTION
**Pull request by cguglielmo**
394805

### CODE REVIEW

File: eclipse-scout-core/src/form/fields/FormFieldValidationResultProvider.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import {fields, FormField, InitModelOf, ObjectModel, ObjectWithType, SomeRequired, Status, ValidationResult} from '../..';
+
+export class FormFieldValidationResultProvider implements ObjectWithType {
+  declare model: FormFieldValidationResultProviderModel;
+  declare initModel: SomeRequired<this['model'], 'field'>;
+  field: FormField;
+  objectType: string;
+
+  init(model: InitModelOf<this>) {
+    $.extend(this, model);
```
Comment (fschinkel) [inline eclipse-scout-core/src/form/fields/FormFieldValidationResultProvider.ts:20]: assert that field is set

File: eclipse-scout-core/src/form/fields/tablefield/TableFieldValidationResultProvider.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        let label = this._getCellLabel(column, row);
+        if (label) {
+          invalidCellLabels.add(label);
+        }
+        if (cellResult.errorStatus?.message) {
+          invalidCellMessages.add(cellResult.errorStatus.message);
+        }
+
+        // Use error status with the highest severity
+        if (!errorStatus) {
```
Comment (fschinkel) [inline eclipse-scout-core/src/form/fields/tablefield/TableFieldValidationResultProvider.ts:77]: have you considered using a multi status that updates itself to the highest severity of its children

File: eclipse-scout-core/test/form/fields/tablefield/TableFieldSpec.ts [hunk 3]
```diff
@@ -215,4 +215,99 @@ describe('TableField', () => {
...
+        expect(tableField.getValidationResult().label).toBe('Table Field: First Col');
+
+        // Invalid columns are included
+        column1.setCellErrorStatus(table.rows[0], Status.error('error'));
+        expect(tableField.getValidationResult().label).toBe('Table Field: First Col, Second Col');
+      });
+    });
+
+    describe('errorStatus', () => {
+      it('contains no message if multiple cells are invalid and have different messages', () => {
```
Comment (fschinkel) [inline eclipse-scout-core/test/form/fields/tablefield/TableFieldSpec.ts:299]: please add another test for the severity

**Review by fschinkel — CHANGES_REQUESTED** [review]

File: eclipse-scout-core/src/form/fields/tablefield/TableFieldValidationResultProvider.ts (hunk 2 above)
Comment (cguglielmo) [reply eclipse-scout-core/src/form/fields/tablefield/TableFieldValidationResultProvider.ts:77]: No, I did not consider that, but as far as I know, there is no MultiStatus in Scout JS, only in Java.

**Pull request merged**